### PR TITLE
chore(renovate): Add renovate config

### PR DESCRIPTION
**Pull request by 0xFelix**
**What this PR does / why we need it**:

Add renovate configuration to automatically update used software versions in e2e-setup and the github actions used by this repository.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

**Review by codingben — APPROVED** [review]
/lgtm

File: renovate.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      "customType": "regex",
+      "fileMatch": ["^hack/e2e-setup.sh$"],
+      "matchStrings": ["KIND_VERSION=\\${KIND_VERSION:-(?<currentValue>.*?)}"],
+      "datasourceTemplate": "github-releases",
+      "packageNameTemplate": "kubernetes-sigs/kind"
+    },
+    {
+      "customType": "regex",
+      "fileMatch": ["^.github/workflows/integration.yml$"],
+      "matchStrings": ["\\sversion: (?<currentValue>.*?)\\n"],
```
Comment (codingben) [inline renovate.json:18]: Will it consider value as `version: v1.2.3` and also `version: 1.2.3`?

Comment (codingben) [reply renovate.json:18]: How did you test this regex? in `regexr.com` or somewhere else?

File: renovate.json [hunk 2]
```diff
@@ -0,0 +1,68 @@
+{
+  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
+  "extends": [":gitSignOff"],
```
Comment (ksimon1) [inline renovate.json:3]: you can add this line https://github.com/kubevirt/kubevirt-tekton-tasks/blob/main/renovate.json#L4
which will create nice dashboard of dependencies like https://github.com/kubevirt/kubevirt-tekton-tasks/issues/496

File: renovate.json (hunk 1 above)
Comment (0xFelix) [reply renovate.json:18]: It would match both, but `v1.2.3` is what's being used.

Yes, or regex101.com.

Comment (codingben) [reply renovate.json:18]: Awesome! :)

**Comment by 0xFelix** [comment]
/hold cancel
/approve

redhat-renovate-bot is active on this repository.

**Pull request merged**